lib/src/Pos/Generator/BlockEvent.hs

```diff
@@ -200,10 +204,11 @@ genBlocksInStructure ::
        , Functor t, Foldable t)
     => AllSecrets
     -> GenesisWStakeholders
+    -> TxpGlobalSettings
     -> Map Path BlockDesc
     -> t Path
     -> RandT g m (t BlundDefault)
-genBlocksInStructure secrets bootStakeholders annotations s = do
+genBlocksInStructure secrets bootStakeholders txpSettings annotations s = do
```

Comment:
Why did you add it as arguments everywhere in this module instead of putting into reader context?

Reply:
makes sense